framework/source/class/qx/lang/Type.js

```diff
@@ -101,7 +101,9 @@ qx.Bootstrap.define("qx.lang.Type",
       // JavaScript null and a null returned from calling DOM.
       // e.q. by document.getElementById("ReturnedNull").
       return (
-        value !== null && (
+        value !== null &&
+        // Check if value is NaN
+        value === value && (
```

Comment:
Number.isNaN may not be supported since that's from recent Ecmascript, but isNaN is supported by all. I believe that the test should include: && ! isNan(value)